cmd/kube-apiserver/app/server.go

```diff
@@ -299,7 +299,15 @@ func Run(s *options.ServerRunOptions) error {
 	}
 
 	admissionControlPluginNames := strings.Split(s.GenericServerRunOptions.AdmissionControl, ",")
-	pluginInitializer := kubeadmission.NewPluginInitializer(client, sharedInformers, apiAuthorizer)
+	var cloudConfig []byte
+
+	if s.CloudProvider.CloudConfigFile != "" {
+		cloudConfig, err = ioutil.ReadFile(s.CloudProvider.CloudConfigFile)
```

Comment:
The admission plugin is ugly hack, I tried to make it use volume plugins in https://github.com/kubernetes/kubernetes/pull/21386 and I failed at that time. With  https://github.com/kubernetes/kubernetes/pull/39425 it could be easier to use volume plugins in API server, however it's 1.7 item (and it's not on my priority list...)

Out of curiosity, why is federation API server cloud-aware? Individual federated API severs can be in different clouds, can't they?